config/src/main/java/org/springframework/security/config/annotation/web/configurers/saml2/Saml2LoginConfigurer.java

```diff
@@ -501,12 +500,7 @@ private <C> C getBeanOrNull(B http, Class<C> clazz) {
 		if (context == null) {
 			return null;
 		}
-		try {
-			return context.getBean(clazz);
-		}
-		catch (NoSuchBeanDefinitionException ex) {
-			return null;
-		}
+		return context.getBeanProvider(clazz).getIfUnique();
```

Comment:
Same comment here about `getIfAvaialble`

Reply:
It's also same `getIfUnique` comment